download_data.py

```diff
@@ -0,0 +1,228 @@
+                        done_percentage = int(100 * downloaded / total_length)
+                        # Update the progress bar
+                        sys.stdout.write(f"\rDownloading {local_filename}: {done_percentage}%")
+                        sys.stdout.flush()
+            if total_length is not None:
+                sys.stdout.write('\n')  # Move the cursor to the next line after download completes
+
+        return True
+    except requests.RequestException as e:
+        print(f"\033[93mWarning: Failed to download {url}. Please download manually.\033[0m")
+        print(f"\033[91mError: {e}\033[0m")
+        return False
+
+
+def extract_tar(file_path, extract_to='.'):
+    """Extract tar file and handle errors."""
+    try:
+        with tarfile.open(file_path) as tar:
+            tar.extractall(path=extract_to)
+        os.remove(file_path)
+    except Exception as e:
+        print(f"\033[93mWarning: Failed to extract {file_path}.\033[0m")
+
+def extract_zip(file_path, extract_to='.'):
+    """Extract zip file and handle errors."""
+    try:
+        with zipfile.ZipFile(file_path, 'r') as zip_ref:
+            zip_ref.extractall(extract_to)
+        os.remove(file_path)
+    except Exception as e:
+        print(f"\033[93mWarning: Failed to extract {file_path}.\033[0m")
+
+def create_dir(path):
+    """Create a directory if it doesn't exist."""
+    os.makedirs(path, exist_ok=True)
+
+def remove_dir(path):
+    """Remove a directory if it exists."""
+    if os.path.exists(path) and os.path.isdir(path):
+        shutil.rmtree(path)
+
+def main():
+    base_path = './data'
+
+    # Download and extract files
+    
+    ## Download firts main file 
+    files_downloaded = []
+    files_skiped = []
+    urls_skiped = []    
+    
+
+    file = 'data_climate_foresight.tar'
+    url  = 'https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/clisight/data_climate_foresight.tar'
+    if download_file(url, file):
+        extract_tar(file)
+        files_downloaded.append(file)        
+    else:
+        files_skiped.append(file)
+        urls_skiped.append(url)
+
+    ## Delete folders and files that we do not need !!!! this should be resolved by removing these files in archive
+    
+    # Remove duplicates directory
+    remove_dir(os.path.join(base_path, 'natural_earth'))
+
+    # Create necessary directories
+    subdirs = ['natural_earth/coastlines',
+               'natural_earth/land',
+               'natural_earth/rivers',
+               'natural_earth/lakes',
+               'natural_hazards',
+               'population']
+    
+    for subdir in subdirs:
+        create_dir(os.path.join(base_path, subdir))
+
+    ## Download and extract other files
+    ###  -------------  coastline
+    file = 'ne_10m_coastline.zip'
+    url  = 'https://naciscdn.org/naturalearth/10m/physical/ne_10m_coastline.zip'
+    if download_file(url, file):
```

Comment:
This looks like something we can do in a loop, given configuration in yaml with list of files and maybe some additional instructions like untaring and so on.

Reply:
Done.
data_sources.yml - yaml config to download data with all links subfolder info ...
[download_data.py](https://github.com/koldunovn/climsight/pull/69/files/5511f6eaceb8ff1af07fae471aeabc4bb12f25b7#diff-bc1a8ac3421ca3c3bbe1a6e4fe4a30a36a153536691319435c8b465d79b9bf73) has a loop over the all these sources from **data_sources.yml**

download_data.sh - should be probably removed from git ?

**data_climate_foresight.tar** at [swift.dkrz.de](https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/clisight/data_climate_foresight.tar) should be updated (it has some trash)

Reply:
Oh yes, we need to figure out what to do with :
WPP2022_Demographic_Indicators_Medium.zip  - can not be downloaded most of the time (some server issues)
and Disaster Location Centroids   (manual downloading is annoying)

Reply:
clean up and moved data_climate_foresight data
WPP2022_Demographic_Indicators_Medium.zip is on dkrz now
pend-gdis-1960-2018-disasterlocations.csv is on dkrz now
citations added to the yml and, text files next to data files